go/consensus/tendermint/seed/seed.go

```diff
@@ -83,6 +99,192 @@ func (srv *seedService) Cleanup() {
 	// No cleanup in particular.
 }
 
+// Synced returns a channel that is closed once synchronization is complete.
+func (srv *seedService) Synced() <-chan struct{} {
+	// Seed is always considered synced.
+	ch := make(chan struct{})
+	close(ch)
+	return ch
+}
+
+// SupportedFeatures returns the features supported by this consensus backend.
+func (srv *seedService) SupportedFeatures() consensus.FeatureMask {
+	return consensus.FeatureMask(0)
+}
+
+// Implements Backend.
+func (srv *seedService) GetStatus(ctx context.Context) (*consensus.Status, error) {
+	status := &consensus.Status{
+		ConsensusVersion: version.ConsensusProtocol.String(),
+		Backend:          api.BackendName,
+		Features:         srv.SupportedFeatures(),
+	}
+
+	// List of consensus peers.
+	tmpeers := srv.p2pSwitch.Peers().List()
+	peers := make([]string, 0, len(tmpeers))
+	for _, tmpeer := range tmpeers {
+		p := string(tmpeer.ID()) + "@" + tmpeer.RemoteAddr().String()
+		peers = append(peers, p)
```

Comment:
No peers in E2E tests when querying seed node status (according to Codecov at least)?

Reply:
Yeah i looked into this earlier today, so there's two things at work here:
- for inbound peers - seed instantly returns the peers and disconnects (as expected)
- for outbound - since we don't set the `"consensus.tendermint.core.external_address"` flag in the tests, nodes connecting to the seed provide `0.0.0.0:<port>` as their external address, the seed then doesn't insert the nodes to its address book and therefore also never makes outbound connections to those

Actually with current test setup seed will only ever return the peers from the genesis and never includes new nodes (but this is enough for all the tests). I also tested this by disabling the `populateAddrBookFromGenesis` in the seed and in that case none of the tests pass. However then configuring the test nodes to provide the correct external addresses (`127.0.0.1:<port>`), things work again correctly :-) - since this is working as expected i didn't commit the test here. However thinking about it more, it might make sense including such a test (disabling automatically including genesis peers in the seeds addressbook) in the suite so that we properly test seed including new peers in its address book.

Reply:
Fixed by configuring `external_addresses` in tests so seed does outbound requests now, also a check is done in the seed test.

Reply:
Also added support for configuring multiple seeds in the e2e tests, and a debug flag for disabling "populating address from genesis", so seed returning new peers is tested as well. Added an E2E test that tests both.